cg/apps/demultiplex/sample_sheet/api.py

```diff
@@ -80,6 +94,78 @@ def validate_sample_sheet(
             file_path=sample_sheet_path, bcl_converter=bcl_converter
         )
 
+    @staticmethod
+    def _is_sample_sheet_from_flow_cell_translatable(flow_cell: FlowCellDirectoryData) -> bool:
+        """
+        Determine if the sample sheet from the flow cell directory is translatable to BCLConvert.
+        """
+        try:
+            flow_cell.run_parameters_path.exists()
+        except FlowCellError:
+            LOG.error(f"Run parameters file for flow cell {flow_cell.full_name} does not exist")
+            return False
+        if not flow_cell.sample_sheet_path.exists():
+            LOG.error(f"Sample sheet for flow cell {flow_cell.full_name} does not exist")
+            return False
+        sample_sheet_content: list[list[str]] = ReadFile.get_content_from_file(
+            file_format=FileFormat.CSV, file_path=flow_cell.sample_sheet_path
+        )
+        if get_sample_type_from_content(sample_sheet_content) is not FlowCellSampleBcl2Fastq:
```

Comment:
How could this happen?

Reply:
This is when you try to translate to BCLConvert a sample sheet that is already in BCLConvert format

Reply:
done! created smaller functions